pkgs/development/compilers/gcc/ng/common/default.nix

```diff
@@ -0,0 +1,236 @@
+          officialRelease
+          version
+          ;
+      })
+      releaseInfo
+      ;
+    inherit (releaseInfo) release_version version;
+    inherit
+      (import ./common-let.nix {
+        inherit
+          lib
+          fetchgit
+          fetchurl
+          release_version
+          gitRelease
+          officialRelease
+          monorepoSrc'
+          version
+          ;
+      })
+      gcc_meta
+      monorepoSrc
+      ;
+    src = monorepoSrc;
+    versionDir =
+      (builtins.toString ../.)
+      + "/${if (gitRelease != null) then "git" else lib.versions.major release_version}";
+    getVersionFile =
+      p:
+      builtins.path {
+        name = builtins.baseNameOf p;
+        path =
+          let
+            patches = args.patchesFn (import ./patches.nix);
+
+            constraints = patches."${p}" or null;
+            matchConstraint =
+              {
+                before ? null,
+                after ? null,
+                path,
+              }:
+              let
+                check = fn: value: if value == null then true else fn release_version value;
+                matchBefore = check lib.versionOlder before;
+                matchAfter = check lib.versionAtLeast after;
+              in
+              matchBefore && matchAfter;
+
+            patchDir =
+              toString
+                (
+                  if constraints == null then
+                    { path = metadata.versionDir; }
+                  else
+                    (lib.findFirst matchConstraint { path = metadata.versionDir; } constraints)
+                ).path;
+          in
+          "${patchDir}/${p}";
+      };
+  };
+in
+makeScopeWithSplicing' {
+  inherit otherSplices;
+  f =
+    gccPackages:
+    let
+      callPackage = gccPackages.newScope (args // metadata);
+
+      mkExtraBuildCommands0 = _: "";
+      mkExtraBuildCommands = _: "";
+    in
+    {
+      stdenv = overrideCC stdenv gccPackages.gcc;
+
+      gcc-unwrapped = callPackage ./gcc {
+        bintools = binutils;
+      };
+
+      libbacktrace = callPackage ./libbacktrace { };
+      libiberty = callPackage ./libiberty { };
+      libsanitizer = callPackage ./libsanitizer { };
+      libquadmath = callPackage ./libquadmath { };
+
+      gfortran-unwrapped = gccPackages.gcc-unwrapped.override {
+        stdenv = overrideCC stdenv buildGccPackages.gcc;
+        langFortran = true;
+      };
+
+      gfortran = wrapCCWith rec {
+        cc = gccPackages.gfortran-unwrapped;
+        libcxx = targetGccPackages.libstdcxx;
+        bintools = binutils;
+        extraPackages = [
+          targetGccPackages.libgcc
+        ];
+        extraBuildCommands =
+          ''
+            echo "-B${targetGccPackages.libgcc}/lib" >> $out/nix-support/cc-cflags
+            echo "-B${targetGccPackages.libssp}/lib" >> $out/nix-support/cc-cflags
+            echo "-B${targetGccPackages.libatomic}/lib" >> $out/nix-support/cc-cflags
+          ''
```

Comment:
This should be transitioned to now newer LLVM does it, I think? By setting a list of `nixSupport.cc-cflags`

Reply:
(Frankly we should also make LLVM < 13 do that too. Is that even a mass rebuild anymore?)

Reply:
Done